signer.go

```diff
@@ -23,6 +23,16 @@ type Signer interface {
 	Sign(rand io.Reader, content []byte) ([]byte, error)
 }
 
+// DigestSigner is an interface for private keys to sign digested COSE signatures.
```

Comment:
this is interface also an integration point with remote KMS? 

any time you have hardware isolated keys, you need a signer interface to request signatures from them.